### PR TITLE
chore: clean up terminal view implemenation

### PR DESCRIPTION
**Pull request by haishengwu-okta**
## Description:

- clean up mock by delete `terminal` objects
- move `terminal` remediation creation into `responseTransformer`
- update tests

## PR Checklist

- [ ] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Issue:

- okta-310251

### CODE REVIEW

File: src/v2/models/AppState.js [hunk 1]
```diff
@@ -87,17 +87,18 @@ export default Model.extend({
...
-    if (!currentViewState) {
-      if (currentFormName) {
-        Logger.warn(`Cannot find view state for form ${currentFormName}. Fall back to terminal state.`);
-      }
+    // didn't expect `remediations` is empty. See `setIonResponse`.
+    const currentViewState = this.get('remediations').filter(r => r.name === currentFormName)[0];
 
-      currentViewState = this.get('terminal');
+    if (!currentViewState ) {
+      Logger.error('Panic!!');
```
Comment (pradeepdewda-okta) [inline src/v2/models/AppState.js:98]: this!! :)

File: playground/mocks/config/responseConfig.js [hunk 2]
```diff
@@ -53,12 +53,11 @@ const idx = {
     // 'identify-with-universal-link',
     // 'select-factor-authenticate',
     // 'select-factor-for-password-recovery',
-    // 'success',
+    'success',
```
Comment (nikhilvenkatraman-okta) [inline playground/mocks/config/responseConfig.js:56]: We should just always  default here to `identify` response. Every time I pull the latest and run `yarn start` I see a different screen that loads up 😆 .

File: src/v2/ion/RemediationConstants.js [hunk 3]
```diff
@@ -36,6 +36,10 @@ const FORMS = {
   DEVICE_APPLE_SSO_EXTENSION: 'device-apple-sso-extension',
   CANCEL_TRANSACTION: 'cancel-transaction',
   LAUNCH_AUTHENTICATOR: 'launch-authenticator',
+
+  // 'terminal` is not ION Form name but only coined in widget
```
Comment (nikhilvenkatraman-okta) [inline src/v2/ion/RemediationConstants.js:40]: 👍

File: src/v2/view-builder/views/TerminalView.js [hunk 4]
```diff
@@ -3,11 +3,6 @@ import BaseView from '../internals/BaseView';
 import BaseForm from '../internals/BaseForm';
 
 const Body = BaseForm.extend({
-  title () {
-    // dont show title for terminal view
-    const msg = this.options.appState.get('terminal').message || {};
```
Comment (nikhilvenkatraman-okta) [inline src/v2/view-builder/views/TerminalView.js:8]: Doesn't this break LEA ? Is the assumption here that API will never send `terminal`? From what I remember LEA API will still send terminal ?

File: playground/mocks/config/responseConfig.js (hunk 2 above)
Comment (haishengwu-okta) [reply playground/mocks/config/responseConfig.js:56]: 😅

File: src/v2/view-builder/views/TerminalView.js (hunk 4 above)
Comment (haishengwu-okta) [reply src/v2/view-builder/views/TerminalView.js:8]: > From what I remember LEA API will still send terminal.

I don't think so after James work. @pradeepdewda-okta correct me wrong.

Comment (nikhilvenkatraman-okta) [reply src/v2/view-builder/views/TerminalView.js:8]: ok I checked my old conversation with James and looks like we removed this from LEA as well. So we should be good!

**Pull request merged**